### DISCUSSION TITLE
GPCode aggregation

### DISCUSSION THREAD

**Discussion by tjrickeard** [opensafely/documentation#132] [Q&A]
We aggregate data at GP level for many of our surveys as vaccines are given in the GP setting. Is there a field we can use as a measure for GPCode?

**Comment by wjchulme** [comment]
To clarify:

Are you asking about the availability of a GP practice identifier? If so, it's possible to return a pseudo identifier for the practice, for example:

```py
    practice_id=patients.registered_practice_as_of(
        index_date,
        returning="pseudo_id",
        ...
    ),
```
This is currently [undocumented](https://docs.opensafely.org/en/latest/study-def-variables/#cohortextractor.patients.registered_practice_as_of) but I've just made a PR to get it updated (https://github.com/opensafely-core/cohort-extractor/pull/455). Proper identifiers at the practice level aren't currently available, [as is explained in the documentation](https://docs.opensafely.org/en/latest/dataset-systmone/#geographic-and-administrative-groupings).

If you're asking about identifying whether vaccines are delivered in the GP setting or elsewhere, then this info doesn't currently make it way to the OpenSAFELY-TPP vaccination tables. You can view the schema for these tables by following second link in the [descriptive metadata section](https://docs.opensafely.org/en/latest/dataset-intro/#descriptive-metadata) section of the documentation.

**Reply by tjrickeard** [reply]
We came across the pseduo_id, but thanks for confirming that. We report on uptake at the lowest level by GP where many of the childhood vaccines are given. The documentation you signpost says MSOA is the lowest level OpenSAFELY can currently extract to. Are there any plans to provide data to GP Practice code level?

**Reply by sebbacon** [reply]
You can reliably use the pseudo id for aggregations etc; it is consistent between all operations within your project workspace.  In terms of more granular mapping to a real-life geography, there are currently no plans to provide anything more precise than MSOA. What is your use-case? Can you give an example of the kind of question you are trying to answer?
